tests/test_generator.py

```diff
@@ -255,55 +243,110 @@ def test_mkdir_file(generator_instance: Generator, test_dir: str):
     assert cmd.command == ["mkdir", "-p", test_dir]
 
 
-def test_copy_file(generator_instance: Generator, fileutils):
-    """Test Generator._copy_files helper function with file"""
-    script = fileutils.get_test_conf_path("sleep.py")
-    files = EntityFiles(copy=script)
-    cmd_list = generator_instance._copy_files(files, generator_instance.root)
-    assert isinstance(cmd_list, CommandList)
-    assert len(cmd_list) == 1
-    assert str(generator_instance.root) and script in cmd_list.commands[0].command
+@pytest.fixture
+def files(fileutils):
+    path_to_files = fileutils.get_test_conf_path(
+        osp.join("generator_files", "easy", "correct/")
+    )
+    list_of_files_strs = sorted(glob(path_to_files + "/*"))
+    yield [pathlib.Path(str_path) for str_path in list_of_files_strs]
 
 
-def test_copy_directory(get_gen_copy_dir, generator_instance: Generator):
-    """Test Generator._copy_files helper function with directory"""
-    files = EntityFiles(copy=get_gen_copy_dir)
-    cmd_list = generator_instance._copy_files(files, generator_instance.root)
-    assert isinstance(cmd_list, CommandList)
-    assert len(cmd_list) == 1
-    assert (
-        str(generator_instance.root)
-        and get_gen_copy_dir in cmd_list.commands[0].command
+@pytest.fixture
+def directory(fileutils):
+    directory = fileutils.get_test_conf_path(
+        osp.join("generator_files", "easy", "correct/")
     )
+    yield [pathlib.Path(directory)]
 
 
-def test_symlink_file(get_gen_symlink_dir, generator_instance: Generator):
-    """Test Generator._symlink_files helper function with file list"""
-    symlink_files = sorted(glob(get_gen_symlink_dir + "/*"))
-    files = EntityFiles(symlink=symlink_files)
-    cmd_list = generator_instance._symlink_files(files, generator_instance.root)
+@pytest.fixture
+def source(request, files, directory):
+    if request.param == "files":
+        return files
+    elif request.param == "directory":
+        return directory
```

Comment:
rather than indirectly setting parameters for this fixture for every test, consider simply parametrizing the fixture itself
```suggestion
@pytest.fixture(params=["files", "directory"])
def source(request):
    yield request.getfixturevalue(request.param)
```